File: python/Van.py
# Inheritance
# Van class inherits properties like vehicleId, vehicleNumber,
# and capacity from the Vehicle class.

from Vehicle import Vehicle


class Van(Vehicle):

    # Van class has its own property as loadCapacity
    def __init__(self, vehicleId, vehicleNumber, capacity, loadCapacity):
        super().__init__(vehicleId, vehicleNumber, capacity)
        self.__loadCapacity = loadCapacity

    # Method overriding
    # Displays van details
    def displayVehicleDetails(self):

        print("\n----------- Van Details -----------")
        print("Vehicle ID :", self.vehicleId)
        print("Vehicle Number :", self.vehicleNumber)
        print("Capacity :", self.capacity)
        print("Load Capacity :", self.__loadCapacity, "kg")

File: java/Van.java
// inheritance 
// van class inherits proprties like vehicleid vehicleNumber and capacity from vehicle class 
public class Van extends Vehicle {
// van class has its own property as a loadCapacity
    private double loadCapacity;

    public Van(int vehicleId, String vehicleNumber, int capacity, double loadCapacity) {

        super(vehicleId, vehicleNumber, capacity);
        this.loadCapacity = loadCapacity;
    }

    @Override
    //displays van details
    public void displayVehicleDetails() {
        System.out.println("\n----------- Van Details -----------");
        System.out.println("Vehicle ID : " + vehicleId);
        System.out.println("Vehicle Number : " + vehicleNumber);
        System.out.println("Capacity : " + capacity);
        System.out.println("Load Capacity : " + loadCapacity + " kg");
    }
}
